modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/CurrencySnapshotAcceptanceManager.scala

```diff
@@ -166,6 +178,31 @@ object CurrencySnapshotAcceptanceManager {
           .getOrElse(acc)
       }
 
+    private def acceptFeeTxs(
+      balances: SortedMap[Address, Balance],
+      maybeTxs: Option[SortedSet[Signed[FeeTransaction]]]
+    ): (SortedMap[Address, Balance], Option[SortedSet[Signed[FeeTransaction]]]) =
```

Comment:
Shouldn't this function return an Either or even throw if the fee transaction cannot be applied?

Reply:
Yeah. I have changed it to throw unexpected state when someone implements fee transactions in a wrong way so balance is negative for example.